### PR TITLE
Check Style configuration to avoid issues with the .editorconfig file…

### PR DESCRIPTION
**Pull request by carlosdlr**
i suggest use the plugin that exists for intellij

https://plugins.jetbrains.com/plugin/1065-checkstyle-idea

for eclipse
https://checkstyle.github.io/eclipse-cs/

and use the microprofile.xml file with the configuration and rules that are already present on the maven check style plugin and adding to the plugin the check style property checkstyle.methodNameFormat with the value that is already defined on the microprofile-fault-tolerance-parent pom.xml  ^_?[a-z][a-zA-Z0-9]*$

to allow to the developers use the plugin that validates the check style rules at compilation time, just importing use the microprofile_checks.xml file

using the plugin, the IDE will start to show the warnings and errors base on the rules imported on the xml file.

github issue
[#190](https://github.com/eclipse/microprofile-fault-tolerance/issues/190)

Signed-off-by: carlos.delarosa <kusanagi12002@gmail.com>

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
@McFoggy is this what you asked for? Can you confirm please?

**Comment by Emily-Jiang** [comment]
@brunobat pls test to see whether this PR fixes the code style in Intellij.

**Comment by brunobat** [comment]
@carlosdlr I've just reviewed your PR on IntelliJ 2017.3.5 ultimate edition.
I tried to import microprofile_checks.xml into the checkstyle-idea plugin and I get an error:

> org.infernus.idea.checkstyle.exception.CheckStylePluginException: <html><b>The Checkstyle rules file could not be parsed.</b><br>Property ${checkstyle.methodNameFormat} has not been set<br>The file has been blacklisted for 60s.</html>
> 	at org.infernus.idea.checkstyle.checker.CheckerFactory.lambda$blacklistAndShowMessage$0(CheckerFactory.java:198)
> ...

image

Also, the microprofile_checks.xml file needs a header. The rat fails on `mvn clean install`

File: microprofile_checks.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (brunobat) [inline microprofile_checks.xml]: Needs header

**Comment by carlosdlr** [comment]
Hello @brunobat

**Comment by brunobat** [comment]
Hi @carlosdlr It still fails to load and my build also fails. 
Do you mind if we have a quick chat about this?

**Comment by brunobat** [comment]
@carlosdlr created this PR for you: https://github.com/carlosdlr/microprofile-fault-tolerance/pull/1

**Comment by carlosdlr** [comment]
Thanks bruno

**Comment by Emily-Jiang** [comment]
@brunobat your PR failed IP-validation
The following users do not have valid ECAs. Please see this page for details, or to sign a ECA.

    brunobat@gmail.com


The following users have invalid Signed-off-by footers. Please use the Signed-Off-By (git commit -s) mechanism to confirm the origin of the submission. The identity in the Signed-off-By must match the Author or the Committer.

    brunobat@gmail.com

**Comment by Emily-Jiang** [comment]
close this one and @carlosdlr will redo one.

**Pull request closed without merging**